### DISCUSSION TITLE
How to change sprite of an entity?

### DISCUSSION THREAD

**Discussion by gmoller** [bevyengine/bevy#2892] [Q&A] [answered]
Hi,

What is the best practice for changing the sprite on an entity? For example, if I want to change the material of an entity created with a SpriteBundle to a different image when it is clicked.

Thanks,
Greg

**Comment by DJMcNab** [comment, marked as answer]
The easiest way would be to change it's `Handle<ColorMaterial>` component to be to a different sprite handle. You would generally store your sprite handles in a resource.

E.g.
```rust
fn my_change_system(sprites: Res<MySprites>, query: Query<&mut Handle<ColorMaterial>, With<MyEntity>>) {
let mut handle = query.single_mut();
*handle = sprites.surprised.clone();
}
```
